src/scales/scale.time.js

```diff
@@ -226,6 +290,22 @@ function determineMajorUnit(unit) {
 	}
 }
 
+function addTick(scale, ticks, time) {
+	if (scale.options.ticks.source !== 'data') {
```

Comment:
Seems a bit odd to have this check in the inner loop.

Reply:
updated